bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -97,26 +97,44 @@ public void propertyChange(PropertyChangeEvent evt) {
                 mm.addMarker(false, this, Severity.Warning, msg, v, a);
             }
 
-			// Function to create markers that report solver intervals.
-			final BiFunction<String, String, Option<Marker>> createSolverMarker =
-				(cacheName, prefix) -> {
+            final Supplier<Union<Interval>> wholeNight = () -> new Union<>(new Interval(v.getSchedule().getStart(), v.getSchedule().getEnd()));
+
+			final Predicate<String> contributes =
+				(cacheName) -> {
 					final Map<Obs, Union<Interval>> c = v.getSchedule().getCache(cacheName);
-					final Union<Interval>       valid = c.getOrDefault(a.getObs(), new Union<>());
+					final Union<Interval> u = c.getOrDefault(a.getObs(), new Union<>());
```

Comment:
I wonder if the default should be the whole night here instead of the empty `Union`.  I believe the cache should have an entry for every observation so it's probably irrelevant.